perf/src/packet.rs

```diff
@@ -20,19 +22,101 @@ pub const PACKETS_PER_BATCH: usize = 64;
 pub const NUM_RCVMMSGS: usize = 64;
 
 #[cfg_attr(feature = "frozen-abi", derive(AbiExample))]
-#[derive(Debug, Default, Clone, Serialize, Deserialize)]
-pub struct PacketBatch {
-    packets: PinnedVec<Packet>,
+#[derive(Debug, Clone)]
+pub struct PacketMutBatch {
```

Comment:
I'm not going to review the streamer changes as part of this, but I'm fairly
sure that this shouldn't exist?

You can create a single large BytesMut::with_capacity(PACKET_SIZE * BATCH_SIZE)
and then split() it.

Reply:
Can you `split()` or `split_at()` `BytesMut` which has only capacity, but no data?

The problem is that we need to pass multiple buffers to `iovec` here:

https://github.com/anza-xyz/agave/blob/49fb51295c1062b6b09e585b2fe0a4676c33d3d4/streamer/src/recvmmsg.rs#L104

I played with splitting non-zeroed `BytesMut` with a given capacity and it doesn't work:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=9a078b4509c7eb1a3e45e207d7ce2975

Is there any trick I don't know about? It would be nice if we can avoid overwriting with zeros and just let our recv code fill the buffers, which is what I'm doing now.